sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1062,8 +1069,10 @@ public Instant getWatermark() {
         return initialWatermark;
       }
 
+      //watermark increases as current_timestamp, in case of latency,
+      //developers need to either provide a customized watermarkFn, or tune Triggers.
       return source.spec.getWatermarkFn() != null
-          ? source.spec.getWatermarkFn().apply(curRecord) : curTimestamp;
```

Comment:
> I set the default watermark to current_timestamp if no user-specified WarermarkFn.

As I mentioned earlier, this works only when curTimestamp is also `now`. Do you agree? Otherwise, most if not all the records could end up as late, which would be a very surprising result for users. I don't think just asking the user to 'set the right watermark fn' as a requirement is a good default. Places a lot of burden even on simple use cases.  

> I guess in kafka0.9, if timestampFn is provided, this watermark cannot guarantee to monotone increasing.
True, but that is not the default. That needed improvement anyway.